libs/x-content/src/main/java/org/opensearch/common/xcontent/XContentType.java

```diff
@@ -127,73 +126,55 @@ public XContent xContent() {
+        return MEDIA_TYPE_PARSER;
+    }
+
     /**
-     * Accepts either a format string, which is equivalent to {@link XContentType#shortName()} or a media type that optionally has
-     * parameters and attempts to match the value to an {@link XContentType}. The comparisons are done in lower case format and this method
-     * also supports a wildcard accept for {@code application/*}. This method can be used to parse the {@code Accept} HTTP header or a
-     * format query string parameter. This method will return {@code null} if no match is found
+     * Accepts a format string, which is most of the time is equivalent to {@link XContentType#subtype()}
+     * and attempts to match the value to an {@link XContentType}.
+     * The comparisons are done in lower case format.
+     * This method will return {@code null} if no match is found
      */
-    public static XContentType fromMediaTypeOrFormat(String mediaType) {
-        if (mediaType == null) {
-            return null;
-        }
-
-        mediaType = removeVersionInMediaType(mediaType);
-        for (XContentType type : values()) {
-            if (isSameMediaTypeOrFormatAs(mediaType, type)) {
-                return type;
-            }
-        }
-        final String lowercaseMediaType = mediaType.toLowerCase(Locale.ROOT);
-        if (lowercaseMediaType.startsWith("application/*")) {
-            return JSON;
-        }
+    public static XContentType fromFormat(String mediaType) {
+        return MEDIA_TYPE_PARSER.fromFormat(mediaType);
+    }
 
-        return null;
+    /**
+     * Attempts to match the given media type with the known {@link XContentType} values. This match is done in a case-insensitive manner.
+     * The provided media type can optionally has parameters.
+     * This method is suitable for parsing of the {@code Content-Type} and {@code Accept} HTTP headers.
+     * This method will return {@code null} if no match is found
+     */
+    public static XContentType fromMediaType(String mediaTypeHeaderValue) {
+        mediaTypeHeaderValue = removeVersionInMediaType(mediaTypeHeaderValue);
+        return MEDIA_TYPE_PARSER.fromMediaType(mediaTypeHeaderValue);
     }
 
     /**
      * Clients compatible with ES 7.x might start sending media types with versioned media type
-     * in a form of application/vnd.opensearch+json;compatible-with=7.
+     * in a form of application/vnd.elasticsearch+json;compatible-with=7.
      * This has to be removed in order to be used in 7.x server.
      * The same client connecting using that media type will be able to communicate with ES 8 thanks to compatible API.
      * @param mediaType - a media type used on Content-Type header, might contain versioned media type.
      *
      * @return a media type string without
      */
     private static String removeVersionInMediaType(String mediaType) {
-        if (mediaType.contains("vnd.opensearch")) {
+        if (mediaType != null && mediaType.contains("vnd.opensearch")) {
```

Comment:
Wondering if we have to lowercase `mediaType` before stripping the version, wdyt?

Reply:
:+1: